pkg/internal/testing/certs/tinyca_test.go

```diff
@@ -0,0 +1,254 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package certs_test
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"math/big"
+	"net"
+	"sort"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gstruct"
+
+	"sigs.k8s.io/controller-runtime/pkg/internal/testing/certs"
+)
+
+var _ = Describe("TinyCA", func() {
+	var ca *certs.TinyCA
+
+	BeforeEach(func() {
+		var err error
+		ca, err = certs.NewTinyCA()
+		Expect(err).NotTo(HaveOccurred(), "should be able to initialize the CA")
+	})
+
+	Describe("the CA certs themselves", func() {
+		It("should be retrievable as a cert pair", func() {
+			Expect(ca.CA.Key).NotTo(BeNil(), "should have a key")
+			Expect(ca.CA.Cert).NotTo(BeNil(), "should have a cert")
+		})
+
+		It("should be usable for signing & verifying", func() {
+			Expect(ca.CA.Cert.KeyUsage&x509.KeyUsageCertSign).NotTo(Equal(0), "should be usable for cert signing")
+			Expect(ca.CA.Cert.KeyUsage&x509.KeyUsageDigitalSignature).NotTo(Equal(0), "should be usable for signature verifying")
+		})
+	})
+
+	It("should produce unique serials among all generated certificates of all types", func() {
```

Comment:
Out of curiosity, does this have any practical relevance? I know the RFC requires it, but is it used for anything other than CRLs?

Reply:
I don't actually remember.  I remember thinking when I wrote TinyCA, "hmm, I should do this semi-correctly and generate unique serials at least", but I don't remember exactly why I thought that